flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManagerTest.java

```
@@ -0,0 +1,65 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition;
+
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for {@link ResultPartitionManager}.
+ */
+public class ResultPartitionManagerTest {
+
+	/**
+	 * Tests that {@link ResultPartitionManager#createSubpartitionView(ResultPartitionID, int, BufferAvailabilityListener)}
+	 * would throw {@link PartitionNotFoundException} if this partition was not registered before.
+	 */
+	@Test
+	public void testThrowPartitionNotFoundException() throws Exception {
+		final ResultPartitionManager partitionManager = new ResultPartitionManager();
+		final ResultPartition partition = PartitionTestUtils.createPartition();
+		try {
+			partitionManager.createSubpartitionView(partition.getPartitionId(), 0, new NoOpBufferAvailablityListener());
+
+			fail("Should throw PartitionNotFoundException for unregistered partition.");
+		} catch (PartitionNotFoundException notFound) {
+			assertEquals(partition.getPartitionId(), notFound.getPartitionId());
+		}
+	}
+
+	/**
+	 * Tests {@link ResultPartitionManager#createSubpartitionView(ResultPartitionID, int, BufferAvailabilityListener)}
+	 * successful if this partition was already registered before.
+	 */
+	@Test
+	public void testCreateViewForRegisteredPartition() {
+		final ResultPartitionManager partitionManager = new ResultPartitionManager();
+		final ResultPartition partition = PartitionTestUtils.createPartition();
+		try {
+			partitionManager.registerResultPartition(partition);
+			partitionManager.createSubpartitionView(partition.getPartitionId(), 0, new NoOpBufferAvailablityListener());
+		} catch (IOException e) {
```

No need to catch the exception. Let it simply bubble up. This is enough to fail the test.

---

That is true. I just thought to give a detail failed message. I could remove to make it simple.